src/app/shared/services/data/app-data.service.spec.ts

```diff
@@ -66,5 +132,22 @@ describe("AppDataService", () => {
   it("should be created", () => {
     expect(service).toBeTruthy();
   });
+
+  it("returns data for a flow", async () => {
+    const data = await service.getSheet("data_list", "flow_a");
+    expect(data.rows?.[0].id).toEqual("a_id1");
+  });
+
+  it("handles flow overrides", async () => {
+    const data = await service.getSheet("data_list", "flow_d");
+    expect(data.rows?.[0].id).toEqual("a_id1");
+  });
+
+  // In the case that flow_b is overridden by flow_c, which in turn is overridden by flow_b,
+  // the loop should be broken after the initial override
```

Comment:
This was what I took to be the expected behaviour, but if we wanted it to allow `flow_a` -> `flow_b` -> `flow_a`, we could change [line 100 of app-data.service.ts](https://github.com/IDEMSInternational/parenting-app-ui/pull/2107/files#diff-d5415ab90f2bbaee6aba223424f7e14faef364f1f635f0db46909bf5bdba7e70R100) from `overrideHistory.push(flow_name)` to `overrideHistory.push(overrideFlowName)`

Reply:
Just to clarify, are we essentially asking whether in the end we should show `flow_a` vs `flow_b`?
I'd expect stopping at `flow_b` - although not sure I fully understand. In either case as long as immediate issue fixed happy for that to be a future update

Reply:
Yes, that's what I was asking. I agree that stopping at `flow_b` seems right, and that seems clearer to me after not thinking about it for a while!